src/GitHub.Exports/Helpers/ThreadingHelper.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Threading.Tasks;
+using ThreadHelper = Microsoft.VisualStudio.Shell.ThreadHelper;
+using GitHub.Extensions;
+using System.Runtime.CompilerServices;
+using System;
+using static Microsoft.VisualStudio.Threading.JoinableTaskFactory;
+using static Microsoft.VisualStudio.Threading.AwaitExtensions;
+
+namespace GitHub.Helpers
+{
+    public interface IAwaitable
+    {
+        IAwaiter GetAwaiter();
+    }
+
+    public interface IAwaiter : INotifyCompletion
+    {
+        bool IsCompleted { get; }
+        void GetResult();
+    }
+
+    public static class ThreadingHelper
+    {
+        /// <summary>
+        /// Switch to the UI thread using ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync
+        /// Auto-disables switching when running in unit test mode
+        /// </summary>
+        /// <returns></returns>
+        public static IAwaitable SwitchToMainThreadAsync()
+        {
+            return Guard.InUnitTestRunner() ?
```

Comment:
That wouldn't work as when you call `await SwitchToMainThreadAsync()` then the calling thread will be captured and continued on at the end of the await unless you remember to call `ConfigureAwait(false)`.

You don't want to have to remember to call `ConfigureAwait(false)` every time its used. Its just one more thing to forget to do and cause yourself issues with calling stuff on the wrong thread.